### ISSUE TITLE
[Question] What program or method do you guys use to import PCB designs to CNCjs?

### ISSUE THREAD

**Issue by mellemhere** [cncjs/cncjs#342]
Hi, first of all, I want to apologise because this is off topic. 

The way I am approaching beeing able to print PCB designs is to explort my desgin as Gerber than open in Flatcam, configure some properties, as cut depth, and than generate my gcode, that I am able to send to the printer using cnc.js.

It seems to me like a huge process and I am not very fond of Flatcam, as it lags and crashes if the Gerber file is too big.

Is my approach optimal? Could I just upload the Gerber file to cncjs and configure cutting path there?

**Comment by kikolobo** [comment]
I use flatCam, with some struggle. I don’t have a clear path for this.

**Comment by AustinSaintAubin** [comment]
CNCjs runs gcode, it is not a post processor or converter.

**Comment by kikolobo** [comment]
Yes, but it would be great if it supported auto leveling. Otherwise I need to use straight end mills instead of v-shapped.

I use an X-Carve so its a big machine to have perfectly level to the nm.

**Comment by mellemhere** [comment]
Yes! Autoleveller is a great missing feature

**Comment by ril3y** [comment]
This is how it works in chilipeppr.

https://github.com/synthetos/TinyG/wiki/ChiliPeppr-PCB-Auto-Level

The code is available as well.  If someone wanted to implement a autolevel
in cncjs I would borrow from john lauers autolevel heavily.

Riley

**Comment by nickm324** [comment]
So was Chillipepper derived from CNCjs or the other way around, in either case they sure look alot alike so I would imagine it would not be that difficulty to implement the autolevel like it is in chillipepper.

**Comment by proffalken** [comment]
https://github.com/pcb2gcode/pcb2gcode and https://github.com/pcb2gcode/pcb2gcodeGUI allow for auto-levelling (although I've not tried it yet!) 

My work pattern is kicad -> pcb2gcode -> cncjs and it works reasonably well.

**Comment by easonke** [comment]
I prefer web based tool chain and use EasyEDA for PCB design, then export to SVG. For now I needs inkscape to convert objects to paths, then feed the svg to jscut to generate gcode. After that it's all cncjs' work.
I am still googling a web based solution for SVG path conversion to replace inkscape.

**Comment by boucz** [comment]
I do use : 
kicad -> carbide3d pcb to gcode (https://carbide3d.com/apps/pcb fantastic free online tool) -> cncjs

But I do need autoleveling ;)

**Comment by cheton** [comment]
The Autolevel feature is now available. See the CNCjs v1.11.0 release for details about the new Autolevel widget support:
https://github.com/cncjs/cncjs/releases/tag/v1.11.0
